crates/key-derivation/src/operator.rs

```diff
@@ -200,31 +201,9 @@ impl Zeroize for OperatorKeys {
             };
         }
 
-        // Zeroize message components
-        message.depth.zeroize();
-        {
-            let fingerprint: &mut [u8; 4] = message.parent_fingerprint.as_mut();
-            fingerprint.zeroize();
-        }
-        message.private_key.non_secure_erase();
-        {
-            let chaincode: &mut [u8; 32] = message.chain_code.as_mut();
-            chaincode.zeroize();
-        }
-        let raw_ptr = &mut message.child_number as *mut ChildNumber;
-        // SAFETY: `message.child_number` is a valid enum variant
-        //          and will not be accessed after zeroization.
-        //          Also there are only two possible variants that will
-        //          always have an `index` which is a `u32`.
-        //          Note that `ChildNumber` does not have the `#[non_exhaustive]`
-        //          attribute.
-        unsafe {
-            *raw_ptr = if message.child_number.is_normal() {
-                ChildNumber::Normal { index: 0 }
-            } else {
-                ChildNumber::Hardened { index: 0 }
-            };
-        }
+        // NOTE: We don't need to zeroize the ed25519 signing key since it is already
+        //       zeroized on drop if we use `ed25519-dalek` with the `zeroize` feature.
+        //       Ref: <https://docs.rs/ed25519-dalek/latest/src/ed25519_dalek/signing.rs.html#665-666>.
```

Comment:
This is not sufficient. It is the case that `SigningKey` is `ZeroizeOnDrop`, but the `Zeroize` trait is intended to capture _manual_ zeroization.